Source/WebCore/dom/Document.cpp

```diff
@@ -2282,6 +2282,31 @@ bool Document::updateStyleIfNeeded()
     return true;
 }
 
+ContentVisibilityForceLayoutScope::ContentVisibilityForceLayoutScope(const Element& context)
+{
+    if (context.document().view()) {
+        context.document().updateRelevancyOfContentVisibilityElements();
+        context.document().updateStyleIfNeeded();
+        if (context.renderer() && context.renderer()->style().skippedContentReason().has_value()) {
+            context.renderer()->setNeedsLayout();
+            m_document = context.document();
+            m_document->view()->layoutContext().forceSkippedContentLayout(true);
+        }
+    }
+}
+
+ContentVisibilityForceLayoutScope::~ContentVisibilityForceLayoutScope()
+{
+    if (m_document && m_document->view())
+        m_document->view()->layoutContext().forceSkippedContentLayout(false);
+}
+
+void Document::updateLayoutForNode(const Element& element)
+{
+    ContentVisibilityForceLayoutScope scope(element);
+    updateLayoutIgnorePendingStylesheets();
+}
```

Comment:
Can we not introduce this new method and instead have a single entry layout entry point that takes an OptionSet ?

It's possible someone might want to update layout not ignoring pending stylesheets, while also taking in account content-visibility. Also, with a single point of entry with options, people will have to think about the `OptionSet<>` and whether they want content-visibility to be taken in account every time they add a new layout.

Reply:
It is a good idea in principle I think. I implemented it, but since the Element also needs to be passed, I am not sure if it is needed in the end, since we can have a bit strange situations where ContentVisiblityForceLayout flag can be passed but not Element, or vice-versa.

However integrating the ContentVisibilityForceLayoutScope helper functionality deeper (right now it is in updateLayout) seems like a good move, ContentVisibilityForceLayoutScope is much smaller now and maybe we do not even need it in the end since it is only used in one place.